cot/src/cache/store.rs

```diff
@@ -0,0 +1,187 @@
+//! Cache store abstractions and implementations.
+//!
+//! This module defines a generic `CacheStore` trait and common types used by
+//! in-memory, file and Redis-backed cache implementations. The main goal is to
+//! provide a simple asynchronous interface for putting, getting, and managing
+//! cached values, optionally with expiration policies.
+
+pub mod memory;
+
+use std::fmt::Debug;
+use std::pin::Pin;
+
+use serde_json::Value;
+use thiserror::Error;
+
+use crate::config::Timeout;
+
+const CACHE_STORE_ERROR_PREFIX: &str = "Cache store error: ";
+
+/// Errors that can occur when interacting with a cache store.
+#[derive(Debug, Error)]
+#[non_exhaustive]
+pub enum CacheStoreError {
+    /// The underlying cache backend returned an error.
+    #[error("{CACHE_STORE_ERROR_PREFIX} Cache store backend error: {0}")]
+    Backend(String),
+    /// Failed to serialize a value for storage.
+    #[error("{CACHE_STORE_ERROR_PREFIX} Serialization error: {0}")]
+    Serialize(String),
+    /// Failed to deserialize a stored value.
+    #[error("{CACHE_STORE_ERROR_PREFIX} Deserialization error: {0}")]
+    Deserialize(String),
+}
+
+/// Convenience alias for results returned by cache store operations.
+pub type CacheStoreResult<T> = Result<T, CacheStoreError>;
+
+/// A generic asynchronous cache interface.
+///
+/// The `CacheStore` trait abstracts over different cache backends. It supports
+/// basic CRUD operations as well as helpers to lazily compute and insert
+/// values, with optional expiration policies.
+pub trait CacheStore: Debug + Send + Sync + 'static {
```

Comment:
Is there any specific reason why `CacheStore` requires `Debug`? If not, I'd argue in favor of removing this requirement - a trait should only require supertraits that if not included, make it useless. In the case of `Debug`, while it's typically implemented for most structures, some people prefer to omit it for better compile times, so it's best not to require it from the users.